### PR TITLE
Instructor: edit student details: email contains raw template literals after resending past links to the new email #8935

### LINKED ISSUE #8935: Instructor: edit student details: email contains raw template literals after resending past links to the new email
**Environment**: 
`master` branch

**Steps to reproduce**
1. Login as instructor
2. Go to sample course/any courses with past sessions
3. Edit a student
4. Change the email of the student to a valid email
5. Click `Yes, save changes and resend links` when the modal box pops up
image

**Expected behaviour**

- No raw template literals at highlighted area:

image

**Actual behaviour**

- Raw template literals at highlighted area:

image

**Additional info**
- Whole process tested in staging server.
- @damithc is this reproducible in the production server? I've found out what's causing the issue.

**New updates**

This case is only reproducible when an instructor changes a student's email who has not joined a course using his/her googleID to another valid email.

### PR DESCRIPTION
**Pull request by tanhengyeow**
Fixes #8935 

**Outline of Solution**

- Added a required variable to be replaced in the template literal. The template literal originated from the template file `studentEmailFragment-courseJoin.html`.

### CODE REVIEW

**Comment by tanhengyeow** [comment]
Ready for review :) @tshradheya

File: src/main/java/teammates/logic/api/EmailGenerator.java [hunk 1]
```diff
@@ -128,7 +128,8 @@ public EmailWrapper generateFeedbackSessionSummaryOfCourse(String courseId, Stud
         String joinFragmentValue = isYetToJoinCourse(student)
                                    ? Templates.populateTemplate(EmailTemplates.FRAGMENT_STUDENT_COURSE_JOIN,
                                            "${joinUrl}", joinUrl,
-                                           "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()))
+                                           "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                                            "${coOwnersEmails}", generateCoOwnersEmailsLine(course.getId()))
```
Comment (tshradheya) [inline src/main/java/teammates/logic/api/EmailGenerator.java]: Indentation? Align with previous line

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 2]
```diff
@@ -63,6 +63,9 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
 
         StudentAttributes student1 = studentsLogic.getStudentForEmail(course.getId(), "student1InCourse1@gmail.tmt");
 
+        // Unregistered student
+        StudentAttributes student2 = studentsLogic.getUnregisteredStudentsForCourse("idOfUnregisteredCourse").get(0);
```
Comment (tshradheya) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: We can avoid the comment and rename the variable to `unregisteredStudent`. Also how about using `getStudentForEmail("idOfUnregisteredCourse", "student1InUnregisteredCourse@gmail.tmt");`

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 3]
```diff
@@ -157,13 +160,22 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
         verifyEmailReceivedCorrectly(emails, student1.email, subject, "/sessionUnpublishedEmailForStudent.html");
         verifyEmailReceivedCorrectly(emails, instructor1.email, subject, "/sessionUnpublishedEmailForInstructor.html");
 
-        ______TS("send summary of all feedback sessions of course email");
+        ______TS("send summary of all feedback sessions of course email to new student. "
+                + "Previous student has joined the course");
```
Comment (tshradheya) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: What do you mean by `previous student` ? How about `edited student` instead?

**Pull request merged**